pgrx/src/spi.rs

```diff
@@ -324,16 +316,16 @@ fn prepare_datum(datum: Option<pg_sys::Datum>) -> (pg_sys::Datum, std::os::raw::
     }
 }
 
-impl<'a> Query for &'a str {
+impl<'conn> Query<'conn> for &str {
     type Arguments = Option<Vec<(PgOid, Option<pg_sys::Datum>)>>;
-    type Result = Result<SpiTupleTable>;
+    type Result = Result<SpiTupleTable<'conn>>;
 
     /// # Panics
     ///
     /// This function will panic if somehow the specified query contains a null byte.
     fn execute(
         self,
-        _client: &SpiClient,
+        _client: &SpiClient<'conn>,
```

Comment:
This is making me realize that I don't understand why Query is its own trait instead of being some kind of extension trait on SpiClient, if not **to** specifically help with this kind of unsound pattern, because it breaks the core lifetime relationship and is implemented the same way in all(?) cases.

However, I am not going to think about that too hard right now as... well, it's a preexisting oddity.